src/MangaBoxParser.ts

```diff
@@ -0,0 +1,208 @@
+    const mangaRootSelector = $(source.mangaRootSelector)
+
+    const image = $(source.mangaThumbnailSelector).attr('src') ?? ''
+
+    const titles = []
+    titles.push(decodeHTML($(source.mangaTitleSelector, mangaRootSelector).text().trim()))
+
+    // Alternative Titles
+    for (const altTitle of $(source.mangaAltTitleSelector, mangaRootSelector)
+        .text()
+        ?.split(/,|;|\//)) {
+        if (altTitle == '') continue
+        titles.push(decodeHTML(altTitle.trim()))
+    }
+
+    const rawStatus = $(source.mangaStatusSelector, mangaRootSelector).text().trim() ?? 'ONGOING'
+    let status = 'ONGOING'
+    switch (rawStatus.toUpperCase()) {
+        case 'ONGOING':
+            status = 'Ongoing'
+            break
+        case 'COMPLETED':
+            status = 'Completed'
+            break
+        default:
+            status = 'Ongoing'
+            break
+    }
+
+    const author = $(source.mangaAuthorSelector, mangaRootSelector)
+        .toArray()
+        .map(x => $(x).text().trim())
+        .join(', ') ?? ''
+
+    const desc = decodeHTML($(source.mangaDescSelector).first().children().remove().end().text().trim())
+
+    let hentai = false
+
+    const tags: Tag[] = []
+    for (const tag of $(source.mangaGenresSelector, mangaRootSelector).toArray()) {
+        const id = $(tag).attr('href')
+        const label = $(tag).text().trim()
+
+        if (!id || !label) continue
+        if (['ADULT', 'SMUT', 'MATURE'].includes(label.toUpperCase())) hentai = true
+
+        tags.push({ id: id, label: label })
+    }
+    const TagSection: TagSection[] = [
+        App.createTagSection({
+            id: '0',
+            label: 'genres',
+            tags: tags.map(t => App.createTag(t)),
+        }),
+    ]
+
+    return App.createSourceManga({
+        id: mangaId,
+        mangaInfo: App.createMangaInfo({
+            image: image,
+            titles: titles,
+            status: status,
+            author: author ? author : 'Unkown',
+            desc: desc,
+            hentai: hentai,
+            tags: TagSection,
+        }),
+    })
+}
+
+export const parseChapters = ($: cheerio.Root, source: MangaBox): Chapter[] => {
+    const chapters: Chapter[] = []
+
+    for (const chapter of $(source.chapterListSelector).toArray()) {
+        const id = $('a', chapter).attr('href') ?? ''
+        if (!id) continue
+
+        const name = decodeHTML($('a', chapter).text().trim())
+        const time = parseDate($(source.chapterTimeSelector, chapter).last().text().trim() ?? '')
+
+        let chapNum = 0
+        const chapRegex = id.match(/(?:chap.*)[-_](\d+\.?\d?)/)
+        if (chapRegex && chapRegex[1]) chapNum = Number(chapRegex[1].replace(/\\/g, '.'))
+
+        chapters.push(App.createChapter({
+            id: id,
+            chapNum: isNaN(chapNum) ? 0 : chapNum,
+            name: name,
+            group: 'Aggragator',
```

Comment:
You can just put an empty string here if not being used.

Reply:
Ah ok, figured it might look empty without something there. I'll make the changes.